runbenchmarks.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import boto3
+import json
+import os
+import time
+import yaml
+from autogluon.bench.cloud.aws.run_deploy import deploy_stack
+from autogluon.bench.frameworks.multimodal.multimodal_benchmark import \
+    MultiModalBenchmark
+from autogluon.bench.frameworks.tabular.tabular_benchmark import \
+    TabularBenchmark
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--config_file', type=str, help='Path to custom config file.')
+
+    args = parser.parse_args()
+    return args
+    
+
+def run_benchmark(configs: dict):
+    if configs["module"] == "multimodal":
+        benchmark = MultiModalBenchmark(benchmark_name=configs["benchmark_name"])
+        git_uri, git_branch = configs["git_uri#branch"].split("#")
+        benchmark.setup(git_uri=git_uri, git_branch=git_branch)
+        benchmark.run(data_path=configs["data_path"])
+        if configs.get("metrics_bucket", None):
+            benchmark.upload_metrics(s3_bucket=configs["metrics_bucket"], s3_dir=f'{configs["module"]}/{benchmark.benchmark_name}')
+    elif configs["module"] == "tabular":
+        benchmark = TabularBenchmark(
+            benchmark_name=configs["benchmark_name"], 
+        )
+        benchmark.setup()
+        benchmark.run(
+            framework=f'{configs["framework"]}:{configs["label"]}',
+            benchmark=configs["amlb_benchmark"],
+            constraint=configs["amlb_constraint"],
+            task=configs["amlb_task"]
+        )
+        if configs["metrics_bucket"] is not None:
+            benchmark.upload_metrics(s3_bucket=configs["metrics_bucket"], s3_dir=f'{configs["module"]}/{benchmark.benchmark_name}')
```

Comment:
The steps involved for AutoMM and Tabular seems to be identical and the only difference is the class to initiate. If so, a cleaner way would be just load the corresponding class from somewhere (e.g. dict mapping from module name to class)

Reply:
The main difference is the parameters to `setup` and `run`, but I extracted them out to a helper class.